sphinx/builders/linkcheck.py

```diff
@@ -312,6 +329,13 @@ def write_entry(self, what, docname, line, uri):
             output.write("%s:%s: [%s] %s\n" % (self.env.doc2path(docname, None),
                                                line, what, uri))
 
+    def write_json_entry(self, data):
+        # type: (dict) -> None
+        data['docpath'] = self.env.doc2path(data['docname'], None)
+        with codecs.open(path.join(self.outdir, 'output.json'), 'a', 'utf-8') as output:  # type: ignore  # NOQA
```

Comment:
Since 3.x, `codecs.open()` is no longer needed. Please use `open()` simply. In addition, last argument `'utf-8'` is also no longer needed.

Reply:
Done. https://docs.python.org/2/howto/pyporting.html#text-versus-binary-data